[taeewonnn] iP

taeewonnn opened the pull request:
# kai.Kai 

> "The end of all knowledge should be service to others." - Cesar Chavez [Source](https://wellbeing.gmu.edu/famous-quotes-on-service-and-well-being/).

## Description of the chatbot
Kai chatbot is glad to help you manage your tasks in a manner that is 
- text-based
- easy to learn
- SUPER friendly to use

All you need to do is,
1. clone the repo from [here](https://github.com/taeewonnn/ip)
2. add your tasks 
3. let it manage for you :grinning:

And it's **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] GUI (coming soon)

try this out on gradle terminal!
```ruby
gradlew run
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

taeewonnn committed:
Renamed, greeted and exited the chat

taeewonnn committed:
Chatbot is able to echo the command the user is typing in

taeewonnn committed:
Made a parsing class so that it can take in inputs and process them as individual commands

taeewonnn committed:
Task class with a taskList collection to hold the task objects, with necessary functionalities present. Duke class able to run with the parsing class and create a list of tasks to do

taeewonnn committed:
Command class to process the input information and use the right functionality. Parsing class making use of Command objects instead of having manual operation with while loop

taeewonnn committed:
Task object has boolean status done. Tasklist with mark and unmark functionality and command class bug fixed. Duke updated with exiting system.

taeewonnn committed:
todo, deadline, event class extending from task class with date and time stored as string for now. parsing class renamed as processor class

taeewonnn committed:
Command is now able to recognise to, deadline and event keywords. tasklist functionalities modified to display task numbers

taeewonnn committed:
A-TextUiTesting. Test using the I/O redirection technique

taeewonnn committed:
Created DukeException class. Processor class not in use.

taeewonnn committed:
handling errors

taeewonnn committed:
delete task

taeewonnn committed:
Delete Processor.java
Duke.java: Change indentations
Command.java: Change indentations
taskList.java: Change indentations

taeewonnn committed:
Add save function

taeewonnn committed:
Add text variation

taeewonnn committed:
Merge branch 'branch-Level-7'

taeewonnn committed:
Add text variation

taeewonnn committed:
Add Date to Deadline class

taeewonnn committed:
Merge branch 'branch-Level-8'

taeewonnn committed:
Add Ui, Storage, Parser

Delete Command.java

Rename taskList.java

taeewonnn committed:
Add text variations

taeewonnn committed:
Merge branch 'branch-A-MoreOOP'

# Conflicts:
#	src/main/java/Duke.java

Delete Duke.java previous main functions

taeewonnn committed:
A-Packages

taeewonnn committed:
Add duke.txt

Show tasks are getting stored in file

taeewonnn committed:
Merge branch 'A-Packages'

taeewonnn committed:
Fix import statements

Import statements are written down with *.

This is a bad way to import.

Let's change it to individual imports

taeewonnn committed:
Add temp.txt

Debugging process

taeewonnn committed:
Merge branch 'add-gradle-support'

taeewonnn committed:
Add gradle support

Build.gradle application was refering to a package seedu that was non existent. Change it to duke.Duke from seedu.duke.Duke

Update duke.txt

taeewonnn committed:
Overwrite temp.txt

Merge commit

taeewonnn committed:
Merge tag 'A-Gradle'

taeewonnn committed:
Add TaskTest.java TaskListTest.java

TaskTest tests for output string and constructor

TaskListTest tests for sucessful add task and failed delete task

taeewonnn committed:
Fix indentation

taeewonnn committed:
Merge branch 'A-JUnit'

taeewonnn committed:
Add header comments to most of the public methods

taeewonnn committed:
Comply with coding standard

taeewonnn committed:
Add find function

taeewonnn committed:
Comply to coding standard

taeewonnn committed:
Merge branch 'A-CodingStandard'

taeewonnn committed:
Merge branch 'A-JavaDoc'

taeewonnn committed:
Merge branch 'branch-Level-9'

taeewonnn committed:
Update coding convention

Name of the variable tasks is not updated in TaskList find method, resolve the conflict

taeewonnn committed:
Update coding convention

Files did not have proper coding convention

Let's change that

taeewonnn committed:
Change package and file name

Chat bot name was still Duke

Let's change that

taeewonnn committed:
Update README.md

taeewonnn committed:
Update README.md

slight changes

hjuntan commented on src/main/java/kai/task/Task.java:
```diff
@@ -0,0 +1,56 @@
+        this.isDone = done;
+    }
+
+    public String outputString() {
```
Consider renaming this into a method that prints the format for writing into files as it a bit ambiguous with the `toString` method

taeewonnn replied:
I definitely agree! I just changed the name of the method to a more explicit and clearer name. Thank you!

hjuntan commented on src/main/java/kai/task/Event.java:
```diff
@@ -0,0 +1,30 @@
+package kai.task;
+
+public class Event extends Task {
+    protected String at;
```
Rename variable `at` into `startDate` or `beginDate`

taeewonnn replied:
Definitely! i changed it to a clearer name

hjuntan commented on src/main/java/kai/util/Parser.java:
```diff
@@ -0,0 +1,61 @@
+        if (inputCommand.equals("list")) {
+            tasklist.listDownTask();
+        } else if (inputCommand.startsWith("mark")) {
+            char pos = inputCommand.charAt(5);
```
What if the user tries to mark the 10th task?

taeewonnn replied:
I did not foresee this potential bug. I made changes to the code. Thank you!

hjuntan commented on src/main/java/kai/util/Parser.java:
```diff
@@ -0,0 +1,61 @@
+import kai.task.TaskList;
+import kai.task.Event;
+
+public class Parser {
```
Add javadoc for each class to identify the role and description of each class

hjuntan commented on src/main/java/kai/Kai.java:
Overall good code quality and standard.

taeewonnn committed:
Implement GUI into chatbot

Let's use javaFX to make a user interface for the users

taeewonnn committed:
Merge branch 'branch-Level-10'

taeewonnn committed:
Comply method name to coding standard

There is a method that outputs the String description of the tasks to
the save file.

The peer review did point out that the method name can be confusing as
the class contains toString() method and the uses can be confusing for
readers.

As a step to make the code more readable, let's refactor all the method
name to a more explicit and clear name. Doing so will make the code much
more readable.

taeewonnn committed:
Fix potential bug

Mark, Unmark, Delete command is only able to read to the 9th tasks in
the list. This may cause a problem when the 10th task is added

Let's fix it so that it is able to make changes to the 10th task and
onwards

taeewonnn committed:
Implement assert statements

Previously, the parser class was not handling most types of errors.
Let's fix that

Storage class can implement assert statements to make sure that the file
is created successfully in createFile method and to make sure that
parsing tasks in the load method can recognize the task types

Let's implement all the features into the different classes

taeewonnn committed:
Improve code quality

There are magic literals found in across the project

Let's replace all the magic literals

There is reverse conditional found

Let's fix that to make it less confusing

taeewonnn committed:
Merge pull request #1 from taeewonnn/branch-A-CodeQuality

Add a branch which improves code quality

taeewonnn committed:
Resolved conflicts in Parser.java and Storage.java

taeewonnn committed:
Merge pull request #2 from taeewonnn/branch-A-Assertions

Implement assert statements

taeewonnn committed:
Fix merge conflict

shayaansultan commented on src/main/java/kai/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy");
```
I like how you have capitalized and finalized the Output Format

shayaansultan commented on src/main/java/kai/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+     * @return New String format.
+     */
+    @Override
+    public String formatStringForSaveFile() {
```
Readable and understandable method name following coding standard 👍

shayaansultan commented on src/main/java/kai/task/Task.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Override
+    public String toString() {
+        return "[" + getIsDoneStatus() + "] " + this.description;
```
Any particular reason why the toString for the app is different from the String to save the file?

shayaansultan commented on src/main/java/kai/task/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+    public String addTask(Task task) {
+        String result;
+        tasks.add(task);
+        result = "Got it. I've added this task:\n"
```
I like how you have divided long lines of code for readability

shayaansultan commented on src/main/java/kai/task/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+     *
+     * @param index
+     */
+    public String markTask(int index) {
```
Could further specify what exactly is being marked i.e. markTaskAsDone for better readability

shayaansultan commented on src/main/java/kai/task/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+     *
+     * @param keyword
+     */
+    public String findTask(String keyword) {
```
Could make this more specific, i.e. findTaskFromString or something for better readability

taeewonnn committed:
Merge branch 'branch-A-Assertions'

taeewonnn committed:
Merge branch 'master' of https://github.com/taeewonnn/ip

taeewonnn committed:
Create help command

It would be good to have a help command that explains every input

Let's implement it

taeewonnn committed:
Merge branch 'C-Help'

taeewonnn committed:
Add Ui.png

A screenshot of the GUI to see how it works should be good

Let's add a screenshot of GUI named Ui.png.

taeewonnn committed:
Update README

There is no comprehensive guide for the users

Let's update README.md file and provide a user guide

taeewonnn committed:
Delete dead code

There are some codes not being used

Let's delete the codes

taeewonnn committed:
Merge branch 'A-UserGuide'

taeewonnn committed:
Improve the code quality

The code in Parser.java is very long and has different abstraction
levels. Javadoc comments can be added and improved in various places

Let's improve all the aspects

taeewonnn committed:
Merge branch 'branch-Finalise'

taeewonnn committed:
Improve code quality

taeewonnn committed:
Create welcome message

No welcome message is shown when the application runs

Let's fix this

taeewonnn committed:
Fix user guide

taeewonnn committed:
Update user guide

taeewonnn committed:
Fix storage issue

taeewonnn committed:
Fix linkage error
